amaranth/hdl/_ast.py

```diff
@@ -1892,6 +1892,60 @@ def __repr__(self):
         return "(cat {})".format(" ".join(map(repr, self.parts)))
 
 
+@final
+class SwitchValue(Value):
+    def __init__(self, test, cases, *, src_loc=None, src_loc_at=0):
+        if src_loc is None:
+            super().__init__(src_loc_at=src_loc_at)
+        else:
+            self.src_loc = src_loc
+        self._test = Value.cast(test)
+        self._cases = []
```

Comment:
This instance variable shouldn't be changing its type in the middle of the constructor.